# Squeezemeta reads analysis

This document outlines the workflow required to perform Squeezemeta Reads (SQM reads) on metagenome sequences for the Australian Microbiome database.

The workflow consists of the following stages:

	A] File preparation and naming
	  1. Preparation of input files
	B] Sequence analysis
	  1. Squeezemeta reads analysis
	C] Post analysis processing
	  1. Add analysis specific information and rename the methods file

### Software used

The following software is used:

	1. Squeezemeta (Tamames and Puente-Sanchez 2019)
	2. Perl 5.32.1
	3. Python 3

Squeezemeta version and database build date used for analysis is provided in the methods file:

	sampleID_MGSD_CSIRO.sqmreads.out.methods.txt

### A] File preparation and naming
#### A-1] Preparation of input files

Data is processed on a per sample basis. A SQM reads input file is generated by concatenating the merged, R1p and R1R2u QC files. An example of the structure of the command used is:

	cat <QC_file_path>/sampleID_*R1p*fastq.gz <QC_file_path>/sampleID_*merged*fastq.gz <QC_file_path>/sampleID_*R1R2u*fastq.gz > <analysis_path>/sampleID_merged-R1p-R1R2u.fastq.gz

The SQM reads tab separated `samples` file is generated with the following information:

	sampleID_SQMreads	sampleID_merged-R1p-R1R2u.fastq.gz 	pair1

### B] Sequence analysis
#### B-1] Squeezemeta reads analysis

The SQM reads script is called using the following command:

	perl <path_to_squeezemeta>/SqueezeMeta/utils/sqm_reads.pl -p sampleID_MGSD_CSIRO.sqmreads -s sampleID.samples -f ./ -t 16

### C] Post analysis processing
#### C-1] Add analysis specific information and rename the methods file

Following analysis, information relating to the build date of the Squeezemeta database used in the analysis is added to methods file by concatenating the `DB_BUILD_DATE` file held in the Squeezemeta database directory. The methods file is then renamed to the format 

	sampleID_MGSD_CSIRO.sqmreads.out.methods.txt

Following SQM reads, output files are renamed with the prefix `sampleID_MGSD_CSIRO.sqmreads` with the following files provided for ingest into the AM processed data portal:

	sampleID_MGSD_CSIRO.sqmreads.out.allreads
	sampleID_MGSD_CSIRO.sqmreads.out.allreads.funcog
	sampleID_MGSD_CSIRO.sqmreads.out.allreads.funkegg
	sampleID_MGSD_CSIRO.sqmreads.out.allreads.mcount
	sampleID_MGSD_CSIRO.sqmreads.out.mappingstat
	sampleID_MGSD_CSIRO.sqmreads.out.methods.txt

### References

J. Tamames and F. Puente-Sanchez. SqueezeMeta, A Highly Portable, Fully Automatic Metagenomic Analysis Pipeline (2019). Frontiers in Microbiology 9, 3349

